lib/net.js

```diff
@@ -425,12 +421,14 @@ Socket.prototype._destroy = function(exception, cb) {
   var self = this;
 
   function fireErrorCallbacks() {
+    var hadSeenError = self._errorEmitted;
+    self._errorEmitted = self._errorEmitted || Boolean(exception);
```

Comment:
Please replace `Boolean(exception)` with `!!exception`, this is what we do everywhere.